From afda257ae652455415d1605392af117e997eb130 Mon Sep 17 00:00:00 2001
From: BuildTools <unconfigured@null.spigotmc.org>
Date: Sat, 8 Dec 2018 21:30:12 -0500
Subject: [PATCH] Hit-Detection-Testing-12


diff --git a/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java b/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
index 965a6a5c..d8997298 100644
--- a/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
+++ b/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
@@ -593,11 +593,19 @@ public class CraftEventFactory {
                             Bukkit.broadcastMessage("called on below 1");
                         }
                     }
-                    else callEvent(event);
+                    else
+                    {
+                        callEvent(event);
+                        Bukkit.broadcastMessage("check 1");
+                    }
                 }
             }
         }
-        else callEvent(event);
+        else
+        {
+            callEvent(event);
+            Bukkit.broadcastMessage("check 2");
+        }
 
         if (!event.isCancelled()) {
             event.getEntity().setLastDamageCause(event);
-- 
2.13.0.windows.1

