From 438638929cb321ca19e80b7b873f00d3b51e3b73 Mon Sep 17 00:00:00 2001
From: BuildTools <unconfigured@null.spigotmc.org>
Date: Sat, 8 Dec 2018 23:04:13 -0500
Subject: [PATCH] Hit-Detection-Testing-15


diff --git a/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java b/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
index 036dccd4..3577c023 100644
--- a/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
+++ b/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
@@ -555,35 +555,35 @@ public class CraftEventFactory {
         {
             if(damager.getBukkitEntity() instanceof CraftPlayer && damagee.getBukkitEntity() instanceof CraftPlayer)
             {
+                CraftPlayer craftDamager = (CraftPlayer) damager.getBukkitEntity();
+                CraftPlayer craftDamagee = (CraftPlayer) damagee.getBukkitEntity();
                 boolean successful = false;
-                Location lineOfSight = ((CraftPlayer)damager.getBukkitEntity()).getEyeLocation();
-                double distanceEye = ((CraftPlayer)damager.getBukkitEntity()).getEyeLocation().distance(((CraftPlayer)damagee.getBukkitEntity()).getLocation());
-                double distanceLocation = ((CraftPlayer)damager.getBukkitEntity()).getLocation().distance(((CraftPlayer)damagee.getBukkitEntity()).getLocation());
+                Location lineOfSight = craftDamager.getEyeLocation();
+                double distanceEye = craftDamager.getEyeLocation().distance(craftDamagee.getLocation());
+                double distanceLocation = craftDamager.getLocation().distance(craftDamagee.getLocation());
                 double distance = distanceEye < distanceLocation ? distanceEye : distanceLocation;
                 if(distance < 5.0d)
                 {
-                    Bukkit.broadcastMessage(((CraftPlayer)damager.getBukkitEntity()).getName() + "'s hit:");
-                    Bukkit.broadcastMessage("distance: " + distance);
                     if(distance >= 0.57d)
                     {
-                        Vector accuracyEye = ((CraftPlayer)damagee.getBukkitEntity()).getEyeLocation().toVector().subtract(lineOfSight.toVector());
-                        Vector accuracyLocation = ((CraftPlayer)damagee.getBukkitEntity()).getLocation().toVector().subtract(lineOfSight.toVector());
-                        double accuracy = accuracyEye.normalize().dot(lineOfSight.getDirection()) > accuracyLocation.normalize().dot(lineOfSight.getDirection()) ? accuracyEye.normalize().dot(lineOfSight.getDirection()) : accuracyLocation.normalize().dot(lineOfSight.getDirection());
-                        Bukkit.broadcastMessage("accuracy: " + accuracy);
+                        Vector vectorEye = craftDamagee.getEyeLocation().toVector().subtract(lineOfSight.toVector());
+                        Vector vectorLocation = craftDamagee.getLocation().toVector().subtract(lineOfSight.toVector());
+                        double accuracyEye = vectorEye.normalize().dot((lineOfSight.getDirection()));
+                        double accuracyLocation = vectorLocation.normalize().dot(lineOfSight.getDirection());
+                        double accuracy = accuracyEye < accuracyLocation ? accuracyEye : accuracyLocation;
                         if(distance >= 4.0d && accuracy >= 0.99d) successful = true;
                         if(distance < 4.0d && distance >= 3.0d && accuracy >= 0.97d) successful = true;
                         if(distance < 3.0d && distance >= 2.0d && accuracy >= 0.94d) successful = true;
                         if(distance < 2.0d && distance >= 1.0d && accuracy >= 0.84d) successful = true;
-                        if(distance < 1.0d && accuracy >= 0.77d) successful = true;
+                        if(distance < 1.0d && distance >= 0.57d && accuracy >= 0.77d) successful = true;
                         Bukkit.broadcastMessage("successful hit: " + successful);
-                        if(successful) callEvent(event);
-                        else return event;
+                        if(!successful) event.setCancelled(true);
                     }
-                    else callEvent(event);
                 }
             }
         }
-        else callEvent(event);
+
+        callEvent(event);
 
         if (!event.isCancelled()) {
             event.getEntity().setLastDamageCause(event);
-- 
2.13.0.windows.1

