From 5def4539849aa9bbc83d6450f4c41613829f5aa5 Mon Sep 17 00:00:00 2001
From: BuildTools <unconfigured@null.spigotmc.org>
Date: Sat, 8 Dec 2018 23:17:54 -0500
Subject: [PATCH] Hit-Detection-Testing-17


diff --git a/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java b/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
index 6a99d511..8d397128 100644
--- a/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
+++ b/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
@@ -571,7 +571,7 @@ public class CraftEventFactory {
                         Vector vectorLocation = craftDamagee.getLocation().toVector().subtract(lineOfSight.toVector());
                         double accuracyEye = vectorEye.normalize().dot((lineOfSight.getDirection()));
                         double accuracyLocation = vectorLocation.normalize().dot(lineOfSight.getDirection());
-                        double accuracy = accuracyEye < accuracyLocation ? accuracyEye : accuracyLocation;
+                        double accuracy = accuracyEye > accuracyLocation ? accuracyEye : accuracyLocation;
                         Bukkit.broadcastMessage("accuracy: " + accuracy);
                         if(distance >= 4.0d && accuracy >= 0.99d) successful = true;
                         if(distance < 4.0d && distance >= 3.0d && accuracy >= 0.97d) successful = true;
-- 
2.13.0.windows.1

