From a966a52727f30cfdf69a37a1578e777ad89d33f9 Mon Sep 17 00:00:00 2001
From: BuildTools <unconfigured@null.spigotmc.org>
Date: Fri, 24 Aug 2018 06:38:50 -0400
Subject: [PATCH] Modified-MeCommand


diff --git a/src/main/java/org/bukkit/command/defaults/MeCommand.java b/src/main/java/org/bukkit/command/defaults/MeCommand.java
index ddcd9ad7..dd85a92e 100644
--- a/src/main/java/org/bukkit/command/defaults/MeCommand.java
+++ b/src/main/java/org/bukkit/command/defaults/MeCommand.java
@@ -3,6 +3,7 @@ package org.bukkit.command.defaults;
 import org.bukkit.Bukkit;
 import org.bukkit.ChatColor;
 import org.bukkit.command.CommandSender;
+import org.bukkit.entity.Player;
 
 public class MeCommand extends VanillaCommand {
     public MeCommand() {
@@ -14,7 +15,11 @@ public class MeCommand extends VanillaCommand {
 
     @Override
     public boolean execute(CommandSender sender, String currentAlias, String[] args) {
-        if (!testPermission(sender)) return true;
+        if (!sender.isOp())
+        {
+            sender.sendMessage(ChatColor.RED + "No permission.");
+            return false;
+        }
         if (args.length < 1)  {
             sender.sendMessage(ChatColor.RED + "Usage: " + usageMessage);
             return false;
-- 
2.13.0.windows.1

